src/coinjoin/client.h

```diff
@@ -178,7 +179,7 @@ class CCoinJoinClientManager
     // TODO: or map<denom, CCoinJoinClientSession> ??
     std::deque<CCoinJoinClientSession> deqSessions GUARDED_BY(cs_deqsessions);
 
-    bool fMixing{false};
+    std::atomic<bool> fMixing{false};
```

Comment:
you should update https://github.com/dashpay/dash/blob/c46306a1df65ecce462c44fff44c207952a655d6/src/coinjoin/client.cpp#L238-L241 to be actually atomic (likely use [compare_exchange](https://en.cppreference.com/w/cpp/atomic/atomic/compare_exchange))?